File: backend/app/schemas/measurement.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class MeasurementBase(BaseModel):
    temperature: Optional[float] = None
    humidity: Optional[float] = None
    soil_moisture: Optional[float] = None
    light_intensity: Optional[float] = None


class MeasurementCreate(MeasurementBase):
    plant_id: int


class Measurement(MeasurementBase):
    id: int
    timestamp: datetime
    plant_id: int

    class Config:
        from_attributes = True

File: frontend/src/types/measurement.ts
export interface Measurement {
  id: number;
  timestamp: string;
  temperature?: number;
  humidity?: number;
  soilMoisture?: number;
  lightIntensity?: number;
  plantId: number;
}

export interface MeasurementCreate {
  plantId: number;
  temperature?: number;
  humidity?: number;
  soilMoisture?: number;
  lightIntensity?: number;
}
